File: practica/C++/5.1.cpp
#include <iostream>

#include <windows.h>
#include <conio.h>

using namespace std;

int main()
{
    SetConsoleCP(1251);
    SetConsoleOutputCP(1251);

    const int CounterRows = 4, CounterColumn = 3;
    int Array[CounterRows][CounterColumn];

    //ввід матриці через рандом
    cout << "Ввід матриці 4х3 (через рандом):" << "\n";
    for(int i = 0; i < CounterRows; i = i + 1)
    {
        for(int j = 0; j < CounterColumn; j = j + 1)
        {
            Array[i][j] = rand() % 10; 
            cout << "\t" << "елемент №" << i + 1 << "-" << j + 1 << ": " << Array[i][j] << "\n";
        }
    }
    

    //вивід матриці
    cout << "\n\n\n" << "Вивід матриці 4х3:" << "\n";
    
    for(int i = 0; i < CounterRows; i = i + 1)
    {
        cout << "\t";
        
        for(int j = 0; j < CounterColumn; j = j + 1)
        {
            cout << Array[i][j] << " ";  
        }

        cout << "\n";
    }


    //сортування рядків
    for(int i = 0; i < CounterRows; i = i + 1) //цикл cтовпців
    {
        //сортування методом бульбашки
        for(int j = 1; j < CounterColumn; j = j + 1)
        {
            for(int k = 0; k < (CounterColumn - j); k = k + 1)
            {
                if(Array[i][k] > Array[i][k + 1])
                {
                    int TempVar = Array[i][k];
                    Array[i][k] = Array[i][k + 1];
                    Array[i][k + 1] = TempVar;
                }
            }
        }
    }

    //вивід матриці
    cout << "\n\n\n" << "Вивід сортованої матриці 4х3 за збільшенням рядка: " << "\n";
    
    for(int i = 0; i < CounterRows; i = i + 1)
    {
        cout << "\t";
        
        for(int j = 0; j < CounterColumn; j = j + 1)
        {
            cout << Array[i][j] << " ";  
        }

        cout << "\n";
    }

    getch();

    return 0;
}

File: practica/pascal/5.1.pas
USES
    crt;

CONST
    CounterRows = 4 - 1;
    CounterColumn = 3 - 1;

VAR
    MyArray : array [0..CounterRows, 0..CounterColumn] of integer;
    TempVar : integer;

BEGIN
    //заповнення матриці
    writeln('Ввід матриці 4х3:');
    
    for i : integer := 0 to CounterRows do
    begin
        for j : integer := 0 to CounterColumn do
        begin
            MyArray[i, j] := random(10); 
            
            writeln('   ', 'елемент №', i + 1, j + 1, ': ', MyArray[i, j]);
        end;
    end;

    writeln();
    writeln(); 
    writeln();   

    //вивід матриці
    writeln('Вивід матриці 4х3:'); 
    
    for i : integer := 0 to CounterRows do
    begin
        write('    ');
        
        for j : integer := 0 to CounterColumn do
        begin
            write(MyArray[i, j], ' ');
        end;

        writeln();
    end;

    writeln();
    writeln();
    writeln();

    //сортування рядків
    for i : integer := 0 to CounterRows do //цикл рядків
    begin
        //сортування методом бульбашки
        for j : integer := 1 to CounterColumn do
        begin
            for k : integer := 0 to (CounterColumn - j) do
            begin
                if (MyArray[i, k] > MyArray[i, k + 1]) then
                begin
                    TempVar := MyArray[i, k];
                    MyArray[i, k] := MyArray[i, k + 1];
                    MyArray[i, k + 1] := TempVar;
                end;
            end;
        end;
    end;

    //вивід матриці
    writeln('Вивід сортованої матриці 4x3 за збільшення рядка: ');
    
    for i : integer := 0 to CounterRows do
    begin
        write('    ');
        
        for j : integer := 0 to CounterColumn do
        begin
            write(MyArray[i, j], ' ');
        end;

        writeln();
    end;

    readkey();
END.